### PR TITLE
br: Add docs for br debug

### PR DESCRIPTION
**Pull request by MoCuishle28**
Signed-off-by: Ubuntu <zhanggaoming028@gmail.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add docs for `br debug`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by shichun-0415** [comment]
@MoCuishle28 Hi, please sign [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=10525). Thanks.
image

**Comment by MoCuishle28** [comment]
seems it shows nothing...
Screenshot from 2022-07-16 11-22-59

**Comment by MoCuishle28** [comment]
> @MoCuishle28 Hi, please sign [Contributor License Agreement](https://cla-assistant.io/pingcap/docs-cn?pullRequest=10525). Thanks. image

seems it shows nothing...
Screenshot from 2022-07-16 11-22-59

File: br/br-usage-backup.md [hunk 1]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
 
+## 备份数据检验
+
+BR 支持在备份完成后，对备份数据进行校验。
+
+### 检查备份数据的完整性
+
+要对备份数据计算校验和，可以使用 `tiup br debug checksum` 命令。
+
+用例：计算 s3 上名为 `backup-data` 的 bucket 下面的 `${prefix}` 前缀目录下的备份的校验和。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:207]:
```suggestion
用例：在 Amazon S3 上名为 `backup-data` 的 bucket 下，计算 `${prefix}` 前缀目录下备份的校验和。
```

File: br/br-usage-backup.md [hunk 2]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+{{< copyable "shell-regular" >}}
```
Comment (shichun-0415) [inline br/br-usage-backup.md:209]: We no longer use copyable.
```suggestion
```

File: br/br-usage-backup.md [hunk 3]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+```shell
+br debug checksum \
+    --storage 's3://backup-data/${prefix}' \
+    --s3.endpoint '${S3-endpoint-URL}' \
+    --log-file checksum.log
+```
+
+### 将备份的 backupmeta 解码为 json 格式的可读文件
+
+在备份完成后，想要将备份的 `backupmeta` 解码为 json 格式的可读文件，查看每个备份文件的键范围、键值对总数等元信息，可以使用 `tiup br debug decode` 命令。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:220]:
```suggestion
在备份完成后，可通过 `tiup br debug decode` 命令将备份的 `backupmeta` 解码为 json 格式的可读文件，从而查看每个备份文件的键范围、键值对总数等元信息。
```

File: br/br-usage-backup.md [hunk 4]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+用例：将 s3 上名为 `backup-data` 的 bucket 下面的 `${prefix}` 前缀目录下的备份的 `backupmeta` 解码为 json 格式的文件 `backupmeta.json`，存储在备份所在的路径为 `s3://backup-data/${prefix}/backupmeta.json`。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:222]:
```suggestion
用例：在 Amazon S3 上名为 `backup-data` 的 bucket 下，将 `${prefix}` 前缀目录下备份的 `backupmeta` 解码为 json 格式的文件 `backupmeta.json`，解码后的文件存储路径为 `s3://backup-data/${prefix}/backupmeta.json`。
```

File: br/br-usage-backup.md [hunk 5]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+{{< copyable "shell-regular" >}}
```
Comment (shichun-0415) [inline br/br-usage-backup.md:224]:
```suggestion
```

File: br/br-usage-backup.md [hunk 6]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+```shell
+br debug decode \
+    --storage 's3://backup-data/${prefix}' \
+    --s3.endpoint '${S3-endpoint-URL}' \
+    --log-file decode-backupmeta.log
+```
+
+将 `backupmeta` 解码为 json 格式的文件后，可以使用 `tiup br debug encode` 命令再将其编码回 `backupmeta`，生成的文件名为 `backupmeta_from_json`。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:233]:
```suggestion
如有需要，你也可以将 json 格式的 `backupmeta` 文件编码回解码前的状态。执行 `tiup br debug encode` 命令，生成的文件名为 `backupmeta_from_json`。
```

File: br/br-usage-backup.md [hunk 7]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+用例：将 s3 上名为 `backup-data` 的 bucket 下面的 `${prefix}` 前缀目录下的备份的 `backupmeta.json` 文件编码为 `backupmeta` 文件，其文件名为 `backupmeta_from_json`，存储在备份所在的路径为 `s3://backup-data/${prefix}/backupmeta_from_json`。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:235]:
```suggestion
用例：在 Amazon S3 上名为 `backup-data` 的 bucket 下，将 `${prefix}` 前缀目录下备份的 `backupmeta.json` 文件编码为 `backupmeta` 文件，编码后的文件名为 `backupmeta_from_json`，存储路径为 `s3://backup-data/${prefix}/backupmeta_from_json`。
```

File: br/br-usage-backup.md [hunk 8]
```diff
@@ -196,6 +196,53 @@ br backup full\
...
+
+{{< copyable "shell-regular" >}}
```
Comment (shichun-0415) [inline br/br-usage-backup.md:237]:
```suggestion
```

**Review by shichun-0415 — APPROVED** [review]

File: br/br-usage-backup.md [hunk 9]
```diff
@@ -196,6 +196,47 @@ br backup full\
...
+```shell
+br debug checksum \
+    --storage 's3://backup-data/${prefix}' \
+    --s3.endpoint '${S3-endpoint-URL}' \
+    --log-file checksum.log
+```
+
+### 将备份的 backupmeta 解码为 json 格式的可读文件
+
+在备份完成后，可通过 `tiup br debug decode` 命令将备份的 `backupmeta` 解码为 json 格式的可读文件，从而查看每个备份文件的键范围、键值对总数等元信息。
```
Comment (IANTHEREAL) [inline br/br-usage-backup.md:218]: 从而查看快照对应的 TSO、每个备份文件的键范围和键值对总数等元信息。

Comment (IANTHEREAL) [reply br/br-usage-backup.md:218]: 如何查看 `查看快照对应的 TSO` 举个例子吧

Comment (MoCuishle28) [reply br/br-usage-backup.md]: 已修改，PTAL

**Comment by IANTHEREAL** [comment]
LGTM

File: br/br-usage-backup.md [hunk 10]
```diff
@@ -196,6 +196,49 @@ br backup full\
 
 BR 支持对备份到 S3 的数据进行 S3 服务端加密 (SSE)。BR S3 服务端加密也支持使用用户自行创建的 AWS KMS 密钥进行加密，详细信息请参考 [BR S3 服务端加密](/encryption-at-rest.md#br-s3-服务端加密)。
 
+## 校验备份数据
+
+使用 BR 完成数据备份后，你可以对备份数据进行校验。
```
Comment (shichun-0415) [inline br/br-usage-backup.md:201]:
```suggestion
使用 BR 完成数据备份后，你可以对备份数据进行校验，包括检查备份数据是否完整，以及通过解码 backupmeta 来查看 TSO 等元信息。
```

**Pull request merged**